src/parser/creative_companion_parser.js

```diff
@@ -14,79 +14,89 @@ import { parserUtils } from './parser_utils';
  */
 export function parseCreativeCompanion(creativeElement, creativeAttributes) {
   const creative = new CreativeCompanion(creativeAttributes);
+  creative.required = creativeElement.getAttribute('required') || null;
 
-  parserUtils
+  creative.variations = parserUtils
     .childrenByName(creativeElement, 'Companion')
-    .forEach(companionResource => {
-      const companionAd = new CompanionAd();
-      companionAd.id = companionResource.getAttribute('id') || null;
-      companionAd.width = companionResource.getAttribute('width');
-      companionAd.height = companionResource.getAttribute('height');
-      companionAd.companionClickTrackingURLTemplates = [];
+    .map(companionResource => {
```

Comment:
Maybe we can make the scenario in which we call `childrenByName` + some array iteration faster, by making children extraction accepting a custom function to apply:
```javascript
function applyToChildrenByName(node, nodeName, fn) {
  const children = [];
  const childNodes = node.childNodes;

  for (const childKey in childNodes) {
    const child = childNodes[childKey];

    if (child.nodeName === name) {
      children.push(fn(child)); // This will save us one iteration
    }
  }
  return children;
}
```

Or
```javascript
function childrenByName(node, nodeName, fn) {
  const hasFunctionToApply = typeof fn === 'function'
  const children = [];
  const childNodes = node.childNodes;

  for (const childKey in childNodes) {
    const child = childNodes[childKey];

    if (child.nodeName === name) {
      hasFunctionToApply ? children.push(fn(child)) : children.push(child);
    }
  }
  return children;
}
```

What do you think about this idea?

Reply:
I think this will make the code less readable in my opinion. Mostly because we would be passing the entire parsing function as a callback :/ I also don't think that iterating twice in this case is a problem (if it would be a large array, maybe...). I don't have a very strong opinion on this though.